go/parquet/schema/reflection_test.go

```diff
@@ -152,6 +152,7 @@ func ExampleNewSchemaFromStruct_logicaltypes() {
 		JSON                  string   `parquet:"logical=json"`
 		BSON                  []byte   `parquet:"logical=BSON"`
 		UUID                  [16]byte `parquet:"logical=uuid"`
+		Float16               [2]byte  `parquet:"logical=float16"`
```

Comment:
can we add a test for an optional one (instead of required) which would use `*[2]byte` as the type?
can we also add a test that uses `float16.Num` as the type?

Reply:
the expected result for the `float16.Num` would be float16 column as per the spec (fixed len binary + float16 logical type). It shouldn't need the `parquet:"logical=float16"` if the type is `float16.Num`. 

You may need to special case the code to check for the float16.Num type to allow it through